python-backend/app/api/documents/expected/resources/expected_document_uploads.py

```diff
@@ -46,50 +45,65 @@ def post(self, expected_document_guid):
             )
 
         expected_document = MineExpectedDocument.find_by_exp_document_guid(expected_document_guid)
+        if not expected_document:
+            return self.create_error_payload(500, f'expected document not found'), 500
```

Comment:
A 500 seems like an incorrect response code for this. A 4-something not-found seems more appropriate to me

Reply:
yea, this should be 400, not 500.